JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -128,13 +130,21 @@ module Microsoft.ApplicationInsights {
             this._applyDeviceContext(envelope, this.device);
             this._applyInternalContext(envelope, this.internal);
             this._applyLocationContext(envelope, this.location);
-            this._applyOperationContext(envelope, this.operation);
             this._applySampleContext(envelope, this.sample);
             this._applyUserContext(envelope, this.user);
+            this._applyOperationContext(envelope, this.operation);
 
             envelope.iKey = this._config.instrumentationKey();
 
-            this._sender.send(envelope);
+            if (this.sample.isSampledIn(envelope)) {
```

Comment:
checked with Vitaly that both metrics and session state types should not be sampled out.